apps/src/templates/census2017/YourSchool.jsx

```diff
@@ -47,10 +50,36 @@ class YourSchool extends Component {
     }
   }
 
+  componentWillReceiveProps(newProps) {
+    if (newProps.schoolDropdownOption !== this.props.schoolDropdownOption) {
```

Comment:
Maybe this is a common pattern but it feels a little unfamiliar to me.  @islemaster?

Reply:
It's not unheard-of.  The common pattern here, if you're building an "uncontrolled" component and you want to set the initial state from props is to use `initial` in the prop name but not the state name.
```
  this.state.schoolDropdownOption = props.initialSchoolDropdownOption;
```
Here, where we're also updating the state when new props are received, I wonder if that makes sense.  Does this mean we're duplicating the `schoolDropdownOption` state, once inside this component and once outside?  Unless there's a particular reason for this, it'd be preferable to only have the prop.

Reply:
:+1: It sounds like we're "duplicating" state but that state is expected to change at different rates in different places, so it's not really duplication.  LGTM!

Reply:
😞 I was hoping you were going to say that there is a simpler way to do this.